### PR TITLE
[DEV APPROVED] #8125 Floating homepage chat

### PR DESCRIPTION
**Pull request by thecountgs**
This implements the floating web chat only on the homepage when there is someone available during business hours.

The chatbox markup was added to the contact footer section of the homepage for code consistency. 

It is set with a hidden class by default and only displays when there is someone available for chat.

I appended my main chatbox container class to the logic that drives the visibility of the chat button already existing in the footer. It uses the existing logic for the footer button to then remove the hidden class and add the link to the chatbox container class.

An application controller method is used to hide the chatbox markup on all internal pages. This gets overridden by the home controller to display only on the home page. 

| Homepage  | Internal Page |
| ------------- | ------------- |
|image | image|

---
This change is [Reviewable](https://reviewable.io/reviews/moneyadviceservice/frontend/1715)

### CODE REVIEW

**Comment by Tr4pSt3R** [comment]
This is an awesome PR!
👍 for the _Backend_ side.

File: app/assets/stylesheets/components/page_specific/_floating_chat.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  @include respond-to($mq-l) {
+    display: block;
+  }
+}
+
+.chat-box__container:hover{
+  text-decoration: none;
+}
```
Comment (johnplayer1982) [inline app/assets/stylesheets/components/page_specific/_floating_chat.scss]: Could you move this block nested inside the ``.chat-box__container`` class above like so:

```
.chat-box__container {
  ...

  &:hover {
    text-decoration: none;
  }
}
```

File: app/assets/stylesheets/components/page_specific/_floating_chat.scss [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  font-size: 1.2rem;
+  margin-bottom: 5px;
+}
+
+.chat-box__icon {
+  margin-left: -35px;	
+}
+
+.chat-box__container--hidden {
+	display: none;
```
Comment (johnplayer1982) [inline app/assets/stylesheets/components/page_specific/_floating_chat.scss]: Indentation is a little off here

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -11,9 +12,13 @@ def contact_panels_homepage?
...
+    true
+  end
+
   private
 
   def interactor
     Core::HomePageReader.new('the-money-advice-service')
   end
-end
+end
```
Comment (johnplayer1982) [inline app/controllers/home_controller.rb]: Please add line break at the end of this document

Comment (thecountgs) [inline app/controllers/home_controller.rb]: Resolved

File: app/assets/stylesheets/components/page_specific/_floating_chat.scss (hunk 2 above)
Comment (thecountgs) [inline app/assets/stylesheets/components/page_specific/_floating_chat.scss]: Strange issue. Resolved

File: app/assets/stylesheets/components/page_specific/_floating_chat.scss (hunk 1 above)
Comment (thecountgs) [inline app/assets/stylesheets/components/page_specific/_floating_chat.scss]: Donezo

**Comment by johnplayer1982** [comment]
:+1:

**Pull request merged**